sunpy/time/time.py

```diff
@@ -215,54 +222,57 @@ def convert_time_str(time_string, **kwargs):
                 break
             if ts is None:
                 continue
-            return datetime.strptime(ts, time_format) + time_delta
+            return Time.strptime(ts, time_format, **kwargs) + time_delta
         except ValueError:
             pass
-    time_string_parse_format = kwargs.pop('_time_string_parse_format', None)
-    if time_string_parse_format is not None:
-        ts, time_delta = _regex_parse_time(time_string,
-                                           time_string_parse_format)
-        if ts and time_delta:
-            return datetime.strptime(ts, time_string_parse_format) + time_delta
-        else:
-            return datetime.strptime(time_string, time_string_parse_format)
+
     # when no format matches, call default fucntion
-    convert_time.dispatch(object)(time_string, **kwargs)
+    return convert_time.dispatch(object)(time_string, **kwargs)
 
 
-def parse_time(time_string, time_format='', **kwargs):
-    """Given a time string will parse and return a datetime object.
+def parse_time(time_string, *, format=None, **kwargs):
+    """Given a time string will parse and return a `astropy.time.Time` object.
     Similar to the anytim function in IDL.
     utime -- Time since epoch 1 Jan 1979
+
     Parameters
     ----------
-    time_string : [ int, float, time_string, datetime ]
-        Date to parse which can be either time_string, int, datetime object.
-    time_format : [ basestring, utime, datetime ]
+    time_string : [ int, float, string, datetime, astropy.time.Time,
+                    numpy.datetime64, pandas.Timestamp ]
+        Time to parse.
+
+    format : [ 'jd', 'mjd', 'decimalyear', 'unix', 'cxcsec', 'gps',
```

Comment:
These are not types they are values, so this line should read ``` format : `str` ``` and the options should be enumerated below. See the fact this dosen't render well currently: https://4707-2165383-gh.circle-artifacts.com/0/root/project/docs/_build/html/api/sunpy.time.parse_time.html#sunpy.time.parse_time

Reply:
In fact the way astropy lists the allowed formats is quite clever: http://docs.astropy.org/en/stable/api/astropy.time.Time.html#astropy.time.Time we should do that.

Reply:
It's also worth noting that in the numpy doc standard it says the following:
> When a parameter can only assume one of a fixed set of values, those values can be listed in braces, with the default appearing first:
```
order : {'C', 'F', 'A'}
    Description of `order`.
```

so we could use our format docstring decorator to auto fill the list of formats.